### PR TITLE
read data from metadata

### PR DESCRIPTION
**Pull request by nikhlagrwl**

### CODE REVIEW

File: lib/chitragupta/util.js [hunk 1]
```diff
@@ -6,30 +6,11 @@ function populateServerData(dataParam) {
...
-  let params;
-  if (indexOfQuestionMark > 0) {
-    endpoint = url.slice(0, indexOfQuestionMark);
-    params = url.slice(indexOfQuestionMark + 1);
-  } else {
-    endpoint = url;
-  }
-
-  data.data.request.method = request.method;
-  data.data.request.endpoint = endpoint;
```
Comment (MitP1997) [inline lib/chitragupta/util.js]: We don't need to remove the entire chunk. We support this + what is passed

File: lib/chitragupta/util.js [hunk 2]
```diff
@@ -6,8 +6,8 @@ function populateServerData(dataParam) {
   const data = dataParam;
   const request = cls.get('request');
   const response = cls.get('response');
-  data.data.request = {};
-  data.data.response = {};
+  data.data.request = data.data.request || {};
+  data.data.response = data.data.response || {};
```
Comment (MitP1997) [inline lib/chitragupta/util.js:10]: all of this needs to be done for process and worker as well

**Review by MitP1997 — COMMENTED** [review]
Also, add examples on usage in README

File: README.md [hunk 3]
```diff
@@ -34,7 +34,7 @@ const winston = require('winston');
 const { Chitragupta } = require('chitragupta');
 
 const jsonLogFormatter = winston.format.printf( ({ level, message, ...meta}) => {
-  const options = {level: level, message: message, meta: meta};
```
Comment (MitP1997) [inline README.md]: data will go in the meta block itself. Also, please highlight the changes in all the examples

File: lib/chitragupta/util.js [hunk 4]
```diff
@@ -46,27 +46,29 @@ function populateServerData(dataParam) {
...
 
 function populateWorkerData(dataParam) {
   const data = dataParam;
-  data.data.worker_name = cls.get('worker_name');
-  data.data.thread_id = cls.get('thread_id');
+  data.data.worker_name = data.data.worker_name || cls.get('worker_name');
+  data.data.thread_id = data.data.thread_id || cls.get('thread_id');
 
-  if (cls.get('job_id')) data.data.job_id = cls.get('job_id');
+  if (cls.get('job_id')) data.data.job_id = data.data.job_id || cls.get('job_id');
```
Comment (MitP1997) [inline lib/chitragupta/util.js]: Please verify the logic here

Comment (nikhlagrwl) [reply lib/chitragupta/util.js]: Tested locally, giving correct output.

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "chitragupta",
-  "version": "1.1.1",
+  "version": "1.1.2",
```
Comment (MitP1997) [inline package.json]:
```suggestion
  "version": "1.2.0",
```

File: lib/chitragupta/util.js [hunk 6]
```diff
@@ -46,17 +46,17 @@ function populateServerData(dataParam) {
...
 }
 
 function populateWorkerData(dataParam) {
   const data = dataParam;
-  data.data.worker_name = cls.get('worker_name');
-  data.data.thread_id = cls.get('thread_id');
+  data.data.worker_name = data.data.worker_name || cls.get('worker_name');
+  data.data.thread_id = data.data.thread_id || cls.get('thread_id');
 
   if (cls.get('job_id')) data.data.job_id = cls.get('job_id');
```
Comment (MitP1997) [inline lib/chitragupta/util.js]: This also needs to be handled, no?

**Review by MitP1997 — APPROVED** [review]

**Pull request merged**